Userland/Libraries/LibCore/System.cpp

```diff
@@ -687,6 +687,15 @@ ErrorOr<off_t> lseek(int fd, off_t offset, int whence)
     return rc;
 }
 
+ErrorOr<void> endgrent()
+{
+    errno = 0;
+    ::endgrent();
```

Comment:
Nit: restore errno when endgrent is successful to maintain the POSIX behavior of not changing errno on success.